front/lib/api/assistant/conversation.ts

```diff
@@ -45,6 +48,73 @@ export async function* postUserMessage(
   | AgentMessageTokensEvent
   | AgentMessageSuccessEvent
 > {
+  const user = auth.user();
+  if (!user) {
+    // TBD
+    throw new Error("User not found");
+  }
+
+  const transaction = await front_sequelize.transaction();
+
+  const lastMessageRank = await AssistantMessage.max<
+    number | null,
+    AssistantMessage
+  >("rank", {
+    where: {
+      assistantConversationId: conversation.id,
+    },
+    transaction,
+  });
+
+  const newUserMessage = await AssistantUserMessage.create(
+    {
+      message: message,
+      userContextUsername: context.username,
+      userContextTimezone: context.timezone,
+      userContextFullName: context.fullName,
+      userContextEmail: context.email,
+      userContextProfilePictureUrl: context.profilePictureUrl,
+      userId: user.id,
+    },
+    {
+      transaction,
+    }
+  );
+  const newMessage = await AssistantMessage.create(
+    {
+      sId: generateModelSId(),
+      rank: lastMessageRank ? lastMessageRank + 1 : 0,
+      assistantConversationId: conversation.id,
+      parentId: null,
+      assistantUserMessageId: newUserMessage.id,
+    },
+    {
+      transaction,
+    }
+  );
+
+  try {
+    await transaction.commit();
```

Comment:
If the `conversation_id` is invalid, this will throw (and rollback the changes).

Reply:
throwing here is fine since the function interface was broken by the caller

Reply:
Why not use a closure which according to @lasryaric is safer as we auto-rollback if we throw somewhere?

Reply:
moved to that